/**:
  ros__parameters:
    run_out:
      detection_method: "Object"  # [-] candidate: Object, ObjectWithoutPath, Points
      use_partition_lanelet: true # [-] whether to use partition lanelet map data
      suppress_on_crosswalk: true # [-] whether to suppress on crosswalk lanelet:
      specify_decel_jerk: false   # [-] whether to specify jerk when ego decelerates
      stop_margin: 2.5            # [m] the vehicle decelerates to be able to stop with this margin
      passing_margin: 1.1         # [m] the vehicle begins to accelerate if the vehicle's front in predicted position is ahead of the obstacle + this margin
      deceleration_jerk: -0.3     # [m/s^3] ego decelerates with this jerk when stopping for obstacles
      detection_distance: 45.0    # [m] ahead distance from ego to detect the obstacles
      detection_span: 1.0         # [m] calculate collision with this span to reduce calculation time
      min_vel_ego_kmph: 3.6       # [km/h] min velocity to calculate time to collision

      detection_area:
        margin_behind: 0.5 # [m] ahead margin for detection area length
        margin_ahead: 1.0  # [m] behind margin for detection area length

      # This area uses points not filtered by vector map to ensure safety
      mandatory_area:
        decel_jerk: -1.2 # [m/s^3] deceleration jerk for obstacles in this area

      # parameter to create abstracted dynamic obstacles
      dynamic_obstacle:
        use_mandatory_area: false # [-] whether to use mandatory detection area
        assume_fixed_velocity:
          enable: false           # [-] If enabled, the obstacle's velocity is assumed to be within the minimum and maximum velocity values specified below
          min_vel_kmph: 0.0       # [km/h] minimum velocity for dynamic obstacles
          max_vel_kmph: 5.0       # [km/h] maximum velocity for dynamic obstacles
        std_dev_multiplier: 1.96  # [-] min and max velocity of the obstacles are calculated from this value and covariance
        diameter: 0.1             # [m] diameter of obstacles. used for creating dynamic obstacles from points
        height: 2.0               # [m] height of obstacles. used for creating dynamic obstacles from points
        max_prediction_time: 10.0 # [sec] create predicted path until this time
        time_step: 0.5            # [sec] time step for each path step. used for creating dynamic obstacles from points or objects without path
        points_interval: 0.1      # [m] divide obstacle points into groups with this interval, and detect only lateral nearest point. used only for Points method

      # approach if ego has stopped in the front of the obstacle for a certain amount of time
      approaching:
        enable: false
        margin: 0.0           # [m] distance on how close ego approaches the obstacle
        limit_vel_kmph: 3.0   # [km/h] limit velocity for approaching after stopping

      # parameters for the change of state. used only when approaching is enabled
      state:
        stop_thresh: 0.01           # [m/s] threshold to decide if ego is stopping
        stop_time_thresh: 3.0       # [sec] threshold for stopping time to transit to approaching state
        disable_approach_dist: 3.0  # [m] end the approaching state if distance to the obstacle is longer than this value
        keep_approach_duration: 1.0 # [sec] keep approach state for this duration to avoid chattering of state transition

      # parameter to avoid sudden stopping
      slow_down_limit:
        enable: true
        max_jerk: -0.7  # [m/s^3] minimum jerk deceleration for safe brake.
        max_acc : -2.0  # [m/s^2] minimum accel deceleration for safe brake.

      # prevent abrupt stops caused by false positives in perception
      ignore_momentary_detection:
        enable: true
        time_threshold: 0.15  # [sec] ignores detections that persist for less than this duration
